src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/TransactionInterceptor.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.companieshouse.overseasentitiesapi.interceptor;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.HandlerMapping;
+import uk.gov.companieshouse.overseasentitiesapi.service.TransactionService;
+import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
+import uk.gov.companieshouse.sdk.manager.ApiSdkManager;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.ERIC_REQUEST_ID_KEY;
+import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.TRANSACTION_ID_KEY;
+
+@Component
+public class TransactionInterceptor implements HandlerInterceptor {
+
+    private final TransactionService transactionService;
+
+    @Autowired
+    public TransactionInterceptor(TransactionService transactionService) {
+        this.transactionService = transactionService;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, @NonNull HttpServletResponse response, @NonNull Object handler) {
+        final Map<String, String> pathVariables =
+                (Map<String, String>) request.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+        final var transactionId = pathVariables.get(TRANSACTION_ID_KEY);
+        String passthroughHeader = request.getHeader(ApiSdkManager.getEricPassthroughTokenHeader());
+
+        var logMap = new HashMap<String, Object>();
+        logMap.put(TRANSACTION_ID_KEY, transactionId);
+        String reqId = request.getHeader(ERIC_REQUEST_ID_KEY);
+        try {
+            ApiLogger.debugContext(reqId, "Getting transaction for request.", logMap);
+
+            final var transaction = transactionService.getTransaction(transactionId, passthroughHeader);
+            ApiLogger.debugContext(reqId, "Transaction retrieved.", logMap);
```

Comment:
Should we add the retrieved transaction to the 'logMap' before this call to 'debug'?

Reply:
I wouldn't say put the whole transaction in as I don't think we should log it out, perhaps adding the transaction_id might be better so we know what transaction was retrieved?

Reply:
Ok. Transaction id already logged added/logged on line 38 so I think we're good here.